sympy/core/add.py

```diff
@@ -234,11 +234,11 @@ def flatten(cls, seq):
         # oo, -oo
         if coeff is S.Infinity:
             newseq = [f for f in newseq if not
-                      (f.is_nonnegative or f.is_real and f.is_finite)]
+                      (f.is_nonnegative or f.is_extended_real and f.is_finite)]
```

Comment:
Perhaps `f.is_finite` should be removed.

Reply:
Agreed.